benchmarks/common.py

```diff
@@ -792,6 +792,71 @@ def decay_batch_exp(self, batch_size, factor=0.5, divisor=2):
             out_batch_size = batch_size - 1
         return max(0, int(out_batch_size))
 
+    def measure_latency_and_memory(self, device, model_name, model_iter_fn, backends):
+        get_peak_memory = lambda: torch.cuda.max_memory_allocated() / 10**9
+        backends.insert(0, "baseline")
+
+        latencies = list()
+        mems = list()
+
+        def mem_experiment(backend):
+            try:
+                # Get the context
+                if backend == "baseline":
+                    ctx = NullContext()
+                else:
+                    ctx = torchdynamo.optimize(backend)
+
+                # Reset and warmup
+                torchdynamo.reset()
+                torch.cuda.empty_cache()
+                t0 = time.perf_counter()
+                with ctx:
+                    model_iter_fn(model, example_inputs)
+                    model_iter_fn(model, example_inputs)
+                    model_iter_fn(model, example_inputs)
```

Comment:
Why do we run thrice between t0 and t1?

Reply:
This is just warmup. t0 and t1 are there to measure compilation cost. Torchscript can take a couple of iterations before it stop recompiling. 

This cost is approximation for compilation latency. It is approximation because it includes model iteration time as well. However, model iteration time typically is in order of ms. Therefore, t1 - t0 is only for flagging really long compilation times (>10 seconds).